frontend/src/app/core/services/match.service.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable } from '@angular/core';
+import { Match } from 'src/app/shared/models/match.model';
+import data from '../../../assets/mock-data.json';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class MatchService {
+
+  constructor() { }
+
+  getSchedule(seasonID: number): Map<number, Match[]> {
```

Comment:
I think this makes sense for the current implementation, but I think we should expand the Season entity to hold 'Number of Rounds'. This will act as validation when adding new matches (i.e. checking that its within the bounds of the number of rounds). It will also mean that we can retrieve all matches, dump them into a Map<number, Match[]>, and then just use the Season.numOfRounds value to iterate from 1 -> numOfRounds. This way we wouldn't need to sort them each time.